dvc/repo/reproduce.py

```diff
@@ -119,14 +119,10 @@ def reproduce(
                     stages.add(stage)
     else:
         for target in targets:
-            stages.update(
-                self.stage.collect(
-                    target,
-                    recursive=recursive,
-                    accept_group=accept_group,
-                    glob=glob,
-                )
+            granular_stages = self.stage.collect_granular(
```

Comment:
Hmm, if we are to change this method, we will effectively make `glob` stop working. We should not do that. 
Ideally we should support `glob` for `collect_granular`. 
But, we could also use `collect_granular` as a fallback in a situation when `glob` is not provided and `collect` returns no results. That way it will be easier but we will need to create issue for supporting `glob` in `collect_granular` ad do it in the future anyway.

Reply:
Second approach is inconsistent, because it would not let user to do `repro` by outs with `glob` option. I'll try to implement `glob` handling logic in `collect_granular`. One more question. There wasn't any tests fails connected to missing `glob` functionality. Should I prepare a test that validates if globbing is possible?

Reply:
@Vonski, this approach also does not work with `accept_group` which supports running multiple `foreach` generated stages, as `collect_granular` does not support that yet (which is yet another thing that introduces an ambiguity for target resolution).

Reply:
Ok, as far as I understand the only acceptable solution is to upgrade `collect_granular` to support globbing and `accept_group`. May I try to do it here?

Reply:
`accept_group` seems to be implemented in the `collect_granular`. Sorry for misleading.

Reply:
@Vonski Sure, go ahead, if you stumble upon any problems, feel free to ping us.

Reply:
Now `collect_granular` handles `glob` also.